### ISSUE TITLE
@fetch-mock/vitest doesn't works with esm package

### ISSUE THREAD

**Issue by kamontat** [wheresrhys/fetch-mock#860]
I try set up fetch-mock/vitest with ESM enabled, it error because no extension found on import path.

```
Error: Cannot find module 'node_modules/.pnpm/@fetch-mock+vitest@0.2.2_vitest@2.1.3/node_modules/@fetch-mock/vitest/dist/esm/vitest-extensions' imported from node_modules/.pnpm/@fetch-mock+vitest@0.2.2_vitest@2.1.3/node_modules/@fetch-mock/vitest/dist/esm/index.js
```

Reproducible repository: https://stackblitz.com/edit/fetch-mock-repro-esm-error?file=README.md&view=editor

The above may seem like a lot of work, but the easier you make it for me to 
recreate the bug, the sooner it'll get fixed.

**Comment by kamontat** [comment]
Add example error message image:

![image](https://github.com/user-attachments/assets/bd6e87d2-47df-4870-b54a-429cce0a5df3)

**Comment by colinfwren** [comment]
Having a similar issue. If I update the import in `@fetch-mock/vitest/dist/esm/index.js` to `import './vitest-extensions.js'` that seems to load the module in and run my tests, not sure if that information will help debug the issue or not

**Comment by wheresrhys** [comment]
Should be fixed in v0.2.3
